src/components/node-list/index.js

```diff
@@ -54,11 +58,26 @@ class NodeList extends React.Component {
 
   /**
    * Return only the results that match the search text
-   * @param {object} results
+   * @param {object} nodes
    */
-  filterResults(results) {
-    return results.filter(node =>
-      this.nodeMatchesSearch(node, this.state.searchValue)
+  filterResults(nodes) {
+    const obj = {};
+    Object.keys(nodes).forEach(key => {
+      obj[key] = nodes[key].filter(node =>
+        this.nodeMatchesSearch(node, this.state.searchValue)
+      );
+    });
+    return obj;
+  }
+
+  /**
+   * Get a list of IDs of the visible nodes
+   * @param {object} filteredNodes
+   */
+  getNodeIDs(filteredNodes) {
+    return Object.keys(filteredNodes).reduce(
+      (nodeIDs, key) => nodeIDs.concat(filteredNodes[key].map(node => node.id)),
+      []
```

Comment:
If there another way to write this or could a comment be penned on whats happening here? Think it looks quite complex with the nested concat + map 😨

Reply:
I've rewritten it to separate it out into different named functions, so that there isn't so much complex logic going on in just a couple of lines

Reply:
👍